lib/stac-auth-proxy/index.ts

```diff
@@ -0,0 +1,138 @@
+import * as cdk from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as apigatewayv2 from "aws-cdk-lib/aws-apigatewayv2";
+import { Construct } from "constructs";
+import { LambdaApiGateway } from "../lambda-api-gateway";
+import { CustomLambdaFunctionProps } from "../utils";
+import * as path from "path";
+
+export class StacAuthProxyLambdaRuntime extends Construct {
+  public readonly lambdaFunction: lambda.Function;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props: StacAuthProxyLambdaRuntimeProps
+  ) {
+    super(scope, id);
+
+    this.lambdaFunction = new lambda.Function(this, "lambda", {
+      runtime: lambda.Runtime.PYTHON_3_13,
```

Comment:
High-level question for **eoAPI-cdk** ... could/should we use a common Python runtime for our lambdas (aka a package-level constant)?

Feels like that could be a _slight_ maintenance burden reduction.

Reply:
I think that's reasonable. It's hard for me to imagine a situation where upgrading Python would break a service.  I'll go even a step further and suggest that it would be reasonable to have a common _default_ but allow end-users to specify the environment.

Reply:
My original comment was brought to you by maintenance of https://earth-search.aws.element84.com/v1, which seemed to run into Lambda env deprecations all the time which always managed to be harder to unpick than we wanted. I figure having a single source of version helps us catch bitrot earlier and (hopefully) make the upgrades less painful.